### PR TITLE
[App Insights] Sync to swagger, bump to 1.0.0-preview-1, rename package *breaking changes*

### PR DESCRIPTION
**Pull request by alexeldeib**
## Description

https://github.com/Azure/azure-rest-api-specs/pull/3771
https://github.com/Azure/azure-rest-api-specs/pull/3084 --> big one
https://github.com/Azure/azure-rest-api-specs/pull/3385
https://github.com/Azure/azure-rest-api-specs/pull/3997
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by alexeldeib** [comment]
I've been refining the data plane swagger for Application Insights and working towards 1.0.0 release for all of our SDKs. This PR rolls up a few changes I've made:

- break Metrics, Events, and Queries into separate operation groups
- minor operation renaming
- move appId parameter from client to method (quality of life; makes sense to new up one client and query with different parameters)
- various type fixes: some timespans need to be strings, some enums should be refined/loosened. 

Adding on to that, this is an SDK to query data plane information -- metrics, logs and events for Azure resources. Application Insights also has numerous clients for *sending* telemetry to Azure, e.g. [**Microsoft.ApplicationInsights**](https://github.com/Microsoft/ApplicationInsights-dotnet).
 After discussing this with App Insights folks and Azure SDK PMs, I'd like to propose renaming this package Microsoft.Azure.ApplicationInsights.Query, to clarify any ambiguity between the two. I've already discussed and begun implementing similar changes with Python, Java, and Node SDKs.  

I've made the relevant code changes already, but I wanted to ask how to go about this procedurally? Is there a precedent for package rename? Could we e.g., leave a note on the Microsoft.Azure.ApplicationInsights package page with a deprecation notice pointing to the newly named package? Would love to get thoughts on this. 

Thanks!

File: src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.Query.csproj [hunk 1]
```diff
@@ -5,15 +5,17 @@
   
   <PropertyGroup>
     <Description>Microsoft Azure Application Insights Data Library</Description>
-    <VersionPrefix>0.9.0.0-preview</VersionPrefix>
-    <AssemblyName>Microsoft.Azure.ApplicationInsights</AssemblyName>
-    <PackageId>Microsoft.Azure.ApplicationInsights</PackageId>
-    <PackageTags>Management.ApplicationInsights;</PackageTags>
+    <VersionPrefix>1.0.0.0</VersionPrefix>
```
Comment (dsgouda) [inline src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.Query.csproj]: versioning should be `1.0.0`
Please rename `VersionPrefix` with `Version`

File: src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.Query.csproj [hunk 2]
```diff
@@ -5,15 +5,17 @@
...
+    <AssemblyName>Microsoft.Azure.ApplicationInsights.Query</AssemblyName>
+    <PackageId>Microsoft.Azure.ApplicationInsights.Query</PackageId>
+    <PackageTags>ApplicationInsights;</PackageTags>
     <PackageReleaseNotes>This is a preview release of the Application Insights Data SDK.</PackageReleaseNotes>
   </PropertyGroup>
   <PropertyGroup>
     <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
-    <RootNamespace>Microsoft.Azure.ApplicationInsights</RootNamespace>
+    <RootNamespace>Microsoft.Azure.ApplicationInsights.Query</RootNamespace>
+    <Version>1.0.0-Preview-1</Version>
```
Comment (dsgouda) [inline src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.Query.csproj]: Please remove this

File: src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.Query.csproj [hunk 3]
```diff
@@ -5,15 +5,17 @@
   
   <PropertyGroup>
     <Description>Microsoft Azure Application Insights Data Library</Description>
-    <VersionPrefix>0.9.0.0-preview</VersionPrefix>
-    <AssemblyName>Microsoft.Azure.ApplicationInsights</AssemblyName>
-    <PackageId>Microsoft.Azure.ApplicationInsights</PackageId>
-    <PackageTags>Management.ApplicationInsights;</PackageTags>
+    <VersionPrefix>1.0.0.0</VersionPrefix>
+    <VersionSuffix>Preview-1</VersionSuffix>
```
Comment (dsgouda) [inline src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.Query.csproj]: Please remove this
Also, is the package stable or preview?
If this is still preview, the version can be bumped to `0.10.0-preview` instead

File: src/SDKs/_metadata/ApplicationInsights_data-plane.txt [hunk 4]
```diff
@@ -1,11 +1,11 @@
-2018-02-27 20:41:17 UTC
+2018-09-23 01:55:41 UTC
 
 1) azure-rest-api-specs repository information
-GitHub user: Azure
-Branch:      master
-Commit:      d122af3086f0b1693b9c506befc497a1f4a8b8e1
+GitHub user: alexeldeib
```
Comment (dsgouda) [inline src/SDKs/_metadata/ApplicationInsights_data-plane.txt]: Is the spec merged?

**Comment by alexeldeib** [comment]
# Incomplete
- [] Update comments for documentation to reflect AppID parameter
- [] Changelog

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
@alexeldeib As such you cannot rename a package, we can hide/deprecate the old package with a message that redirects to a new package and then publish whichever new packages are needed. Please start an email thread if you need further details

File: src/SDKs/_metadata/ApplicationInsights_data-plane.txt (hunk 4 above)
Comment (alexeldeib) [reply src/SDKs/_metadata/ApplicationInsights_data-plane.txt]: There's an open PR which changes an enum to modelAsString. Rest of the changes are merged.

Comment (alexeldeib) [reply src/SDKs/_metadata/ApplicationInsights_data-plane.txt]: The same open PR contains the autorest config to change namespace to Microsoft.Azure.ApplicationInsights.Query

Comment (alexeldeib) [reply src/SDKs/_metadata/ApplicationInsights_data-plane.txt]: I can regenerate after it's merged. The same PR contains the namespace change.

File: src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.Query.csproj (hunk 2 above)
Comment (alexeldeib) [reply src/SDKs/ApplicationInsights/DataPlane/ApplicationInsights/Microsoft.Azure.ApplicationInsights.Query.csproj]: fixed

**Comment by alexeldeib** [comment]
@dsgouda confirming, for preview version you want 0.10.0-preview **not**  1.0.0-preview? I made it 0.10.0-preview currently.

File: src/SDKs/ApplicationInsights/DataPlane/Data.ApplicationInsights.Tests/Data.ApplicationInsights.Tests.csproj [hunk 5]
```diff
@@ -14,7 +14,9 @@
   </ItemGroup>
   <ItemGroup>
```
Comment (dsgouda) [inline src/SDKs/ApplicationInsights/DataPlane/Data.ApplicationInsights.Tests/Data.ApplicationInsights.Tests.csproj]: Please replace `VersionPrefix` with `Version`
Also add `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>` to the same `PropertyGroup`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good apart from the one comment

File: src/SDKs/ApplicationInsights/DataPlane/Data.ApplicationInsights.Tests/Data.ApplicationInsights.Tests.csproj (hunk 5 above)
Comment (alexeldeib) [reply src/SDKs/ApplicationInsights/DataPlane/Data.ApplicationInsights.Tests/Data.ApplicationInsights.Tests.csproj]: Fixed

**Review by dsgouda — APPROVED** [review]
LGTM subject to CIs passing

**Comment by alexeldeib** [comment]
@dsgouda Bump since CI is green?

**Pull request merged**